starter-code/code.js

```diff
@@ -0,0 +1,151 @@
+//Iteration 1 The Office
+
+var opinions = [ "This is the best job ever",
+"Satisfied",
+"At least I get paid",
+"I'm looking for another job",
+"I don't want to answer"];
+
+var employeeSatisfaction = function () {
+
+	var randOpinions = function (){
+		return _.sample(opinions);
+	}
+
+	var fillNewArray = function(){
+
+		return _.times(10, randOpinions);
+
+	}
+
+	var fillNewArray2 = function(){
+		return _.times(5, fillNewArray)
+	}
+
+	return fillNewArray2();
+};
+
+console.log(employeeSatisfaction());
+
+//Iteration2 harry Potter's Birthday
+
+var birthdays = ["Severus Snape", "9 January", "Arthur Weasley", "6 February",
+"Ron Weasley", "1 March", "Remus Lupin", "10 March",
+"Fred & George Weasley", "1 April", "Pomona Sprout", "15 May",
+"Draco Malfoy", "5 June", "Dobby", "28 June",
+"Neville Longbottom", "30 July", "Harry Potter", "31 July",
+"Ginevra Weasley", "11 August", "Percy Weasley ", "22 August",
+"Hermione Granger", "19 September", "Minerva McGonagall",
+"4 October", "Filius Flitwick", "17 October", "Molly Weasley",
+"30 October", "Bill Weasley", "29 November", "Rubeus Hagrid",
+"6 December", "Charlie Weasley", "12 December"];
+
+var moreBirthdays = ["Lily Evans", "30 January", "James Potter", "27 March",
+"Dudley Dursley", "30 June", "Tom Riddle", "31 December"];
+
+function groupArray(array){
+	return _.chunk(array, 2)
+}
+
+function addToArray(array1, array2){
+	return _.concat(groupArray(array1), groupArray(array2))
+}
+
+console.log(addToArray(birthdays, moreBirthdays));
+
+
+//Iteration3 The Password Problem 
+
+//exercise 1
+var goodPsswd = "1234567890";
+var badPsswd = "1123456";
+
+var noRepeatChar = function (pwd){
+	var result = _.uniq(pwd).length === pwd.length ? "your password is fine" : "your password has a duplicate";
+
+	return result;
+}
+console.log("Good unique Password: ", noRepeatChar(goodPsswd));
+console.log("Bad unique Password: ", noRepeatChar(badPsswd));
+
+//exercise 2
+var goodPsswd = "1234567890";
+var badPsswd = "1a234567890";
+
+var onlyNumbers = function password2(pwd){
+	var result = _.toNumber(pwd) ? "your password is fine" : "your password has a character";
```

Comment:
in order to check if some string includes some specific kind of digit or not you can use regular expressions